### PR TITLE
feat(replays): added log level filtering to the replay details console tab

### LINKED ISSUE #34659: Allow console messages to be filterable by log levels (e.g. warning, error, info)

### PR DESCRIPTION
**Pull request by Danondso**
Closes #34659 

image

image

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: static/app/views/replays/detail/console/index.tsx [hunk 1]
```diff
@@ -15,21 +16,45 @@ interface Props {
...
-      <ConsoleSearch onChange={handleSearch} />
+      <ConsoleFilters>
+        <CompactSelect
+          triggerProps={{
+            prefix: t('Log Level'),
+          }}
+          multiple
+          options={getDistinctLogLevels(breadcrumbs).map(breadcrumbLogLevel => ({
+            value: breadcrumbLogLevel,
+            label: breadcrumbLogLevel as string,
```
Comment (ryan953) [inline static/app/views/replays/detail/console/index.tsx:49]: i don't think `as string` is the right fix

Either convert to string explicitly `label: String(breadcrumbLogLevel),`

Remember that `String(undefined) === "undefined"`. So the fix might really be to make `breadcrumbLogLevel` be typed as `string` and not `string | undefined` at its source.

Comment (Danondso) [reply static/app/views/replays/detail/console/index.tsx]: I put a type of string on the set of log levels, BreadcrumbLevelType doesn't have undefined as a possible value so the issue was just local to my implementation.

File: static/app/views/replays/detail/console/index.tsx [hunk 2]
```diff
@@ -47,6 +72,15 @@ function Console({breadcrumbs}: Props) {
   );
 }
 
+const ConsoleFilters = styled('div')`
+  display: grid;
+  gap: ${space(1)};
+  grid-template-columns: max-content 1fr;
```
Comment (billyvg) [inline static/app/views/replays/detail/console/index.tsx:78]:
```suggestion
  grid-template-columns: max-content auto;
```

File: static/app/views/replays/detail/console/index.tsx [hunk 3]
```diff
@@ -47,6 +72,15 @@ function Console({breadcrumbs}: Props) {
...
+  width: 100%;
+  margin-bottom: ${space(1)};
+  margin-top: ${space(1)};
```
Comment (billyvg) [inline static/app/views/replays/detail/console/index.tsx:81]: We only need top margin when it's on a smaller screen size

File: static/app/views/replays/detail/console/index.tsx [hunk 4]
```diff
@@ -47,6 +72,15 @@ function Console({breadcrumbs}: Props) {
   );
 }
 
+const ConsoleFilters = styled('div')`
+  display: grid;
+  gap: ${space(1)};
+  grid-template-columns: max-content 1fr;
+  width: 100%;
```
Comment (billyvg) [inline static/app/views/replays/detail/console/index.tsx:79]: I don't think this is needed?

**Review by billyvg — COMMENTED** [review]
The label looks a bit funny when nothing is selected?

**Comment by billyvg** [comment]
We should add a placeholder to the search input as well.

File: static/app/views/replays/detail/console/index.tsx (hunk 3 above)
Comment (billyvg) [reply static/app/views/replays/detail/console/index.tsx:81]: @Danondso I think you missed these comments :D

**Review by billyvg — APPROVED** [review]

**Pull request merged**